test/dynamo/test_hooks.py

```diff
@@ -889,6 +889,35 @@ def forward(self, x):
         self.assertEqual(ref, res)
         self.assertEqual(cnts.frame_count, 1)
 
+    def test_global_module_forward_pre_hook(self):
+        class Mod(torch.nn.Module):
+            def forward(self, x):
+                return x - 1
+
+        counter = 0
+
+        def hook(mod, args):
+            nonlocal counter
+            counter += 1
+            return args
+
+        mod = Mod()
+        torch.nn.modules.module.register_module_forward_pre_hook(hook)
```

Comment:
Does this pollute the state of all the other tests? Can we do a try-finally where we deregister the hook in the finally?

Reply:
That is a _good_ call lol, updated.